classes/ActionScheduler_WPCLI_Scheduler_command.php

```diff
@@ -54,23 +58,23 @@ public function run( $args, $assoc_args ) {
 			$runner = new ActionScheduler_WPCLI_QueueRunner( null, null, $cleaner );
 
 			// Determine how many tasks will be run in the first batch.
-			$total = $runner->setup( $batch, $hooks, $group, $force );
+			$total = $runner->setup( $batch, $hooks, $group, $force, $print_timestamp );
 
 			// Run actions for as long as possible.
 			while ( $total > 0 ) {
-				$this->print_total_actions( $total );
+				$this->print_total_actions( $total, $print_timestamp );
 				$actions_completed += $runner->run();
 				$batches_completed++;
 
 				// Maybe set up tasks for the next batch.
-				$total = ( $unlimited || $batches_completed < $batches ) ? $runner->setup( $batch, $hooks, $group, $force ) : 0;
+				$total = ( $unlimited || $batches_completed < $batches ) ? $runner->setup( $batch, $hooks, $group, $force, $print_timestamp ) : 0;
 			}
 		} catch ( Exception $e ) {
-			$this->print_error( $e );
+			$this->print_error( $e, $print_timestamp );
 		}
 
-		$this->print_total_batches( $batches_completed );
-		$this->print_success( $actions_completed );
+		$this->print_total_batches( $batches_completed, $print_timestamp );
+		$this->print_success( $actions_completed, $print_timestamp );
```

Comment:
A separate function exclusively for prepending the date string will help reduce a lot of the code being added in the `print_*` methods. Especially if the `$print_timestamp` flag can be stored as a property on the class and accessed via that method.

Reply:
Agreed. I've optimized this in 814b6b2df56baadaa721061a8b9cbafa182f1a1a.